<?xml version="1.0" ?>
<robot name="seed_hand">
	<link name="right_wrist:11">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/right_wrist_iam_81504746.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.010971 -0.017043 0.01141"/>
			<material name="right_wrist:11_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/right_wrist_iam_81504746.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.010971 -0.017043 0.01141"/>
		</collision>
		<inertial>
			<mass value="0.028397"/>
			<inertia ixx="0.000043" ixy="-0" ixz="0.000004" iyy="0.000058" iyz="-0.000007" izz="0.000029"/>
			<origin rpy="0 0 0" xyz="-0.01062 -0.034715 0.014091"/>
		</inertial>
	</link>
	<link name="right_palm:11">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/right_palm_iam_583b22df.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.022933 0.001905 -0.017234"/>
			<material name="right_palm:11_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/right_palm_iam_583b22df.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.022933 0.001905 -0.017234"/>
		</collision>
		<inertial>
			<mass value="0.005332"/>
			<inertia ixx="0.000015" ixy="0.000001" ixz="-0.000004" iyy="0.000017" iyz="0.000004" izz="0.000006"/>
			<origin rpy="0 0 0" xyz="-0.029511 0.002252 -0.023968"/>
		</inertial>
	</link>
	<joint name="r_palm_x" type="revolute">
		<parent link="right_wrist:11"/>
		<child link="right_palm:11"/>
		<origin rpy="-0.433097 0 0" xyz="0.012764 -0.017607 0.011254"/>
		<axis xyz="1 0 0"/>
		<limit effort="30" lower="0" upper="0.872665" velocity="1"/>
	</joint>
	<link name="finger_segment:22">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006681 -0.015507 -0.007416"/>
			<material name="finger_segment:22_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006681 -0.015507 -0.007416"/>
		</collision>
		<inertial>
			<mass value="0.000009"/>
			<inertia ixx="0" ixy="0" ixz="-0" iyy="0" iyz="0" izz="0"/>
			<origin rpy="0 0 0" xyz="0.004909 0.000556 -0.005751"/>
		</inertial>
	</link>
	<joint name="r_ringfinger_base_x" type="revolute">
		<parent link="right_palm:11"/>
		<child link="finger_segment:22"/>
		<origin rpy="-0.569851 -0.062904 -3.141593" xyz="-0.035879 0.006226 -0.040569"/>
		<axis xyz="-0.998022 0.00962907 0.0621208"/>
		<limit effort="30" lower="0" upper="1.309" velocity="1"/>
	</joint>
	<link name="finger_segment:12">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006181 -0.015507 -0.007416"/>
			<material name="finger_segment:12_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006181 -0.015507 -0.007416"/>
		</collision>
		<inertial>
			<mass value="0.000009"/>
			<inertia ixx="0" ixy="0" ixz="-0" iyy="0" iyz="0" izz="0"/>
			<origin rpy="0 0 0" xyz="0.005409 0.000556 -0.005751"/>
		</inertial>
	</link>
	<joint name="r_ringfinger_1st_x" type="revolute">
		<parent link="finger_segment:22"/>
		<child link="finger_segment:12"/>
		<origin rpy="-0.095245 0 0" xyz="-0.0005 0 -0.016"/>
		<axis xyz="-0.998022 0.00962907 0.0621208"/>
		<limit effort="30" lower="0" upper="1.309" velocity="1"/>
	</joint>
	<link name="fingertip:12">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/fingertip_ipt_8417b8a9.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.005514 0.000374 -0.010171"/>
			<material name="fingertip:12_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/fingertip_ipt_8417b8a9.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.005514 0.000374 -0.010171"/>
		</collision>
		<inertial>
			<mass value="0.000008"/>
			<inertia ixx="0" ixy="-0" ixz="-0" iyy="0" iyz="-0" izz="0"/>
			<origin rpy="0 0 0" xyz="0.005013 0.000408 -0.010216"/>
		</inertial>
	</link>
	<joint name="r_ringfinger_2nd_x" type="revolute">
		<parent link="finger_segment:12"/>
		<child link="fingertip:12"/>
		<origin rpy="-0.109614 0 0" xyz="0 0 -0.016"/>
		<axis xyz="-0.998022 0.00962907 0.0621208"/>
		<limit effort="30" lower="0" upper="1.309" velocity="1"/>
	</joint>
	<link name="finger_segment:21">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006681 -0.015507 -0.007416"/>
			<material name="finger_segment:21_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006681 -0.015507 -0.007416"/>
		</collision>
		<inertial>
			<mass value="0.000009"/>
			<inertia ixx="0" ixy="0" ixz="-0" iyy="0" iyz="0" izz="0"/>
			<origin rpy="0 0 0" xyz="0.004909 0.000556 -0.005751"/>
		</inertial>
	</link>
	<joint name="r_thumb_base_x" type="revolute">
		<parent link="right_palm:11"/>
		<child link="finger_segment:21"/>
		<origin rpy="1.915504 0 0" xyz="-0.0275 0.0199 -0.00842"/>
		<axis xyz="-1 0 0"/>
		<limit effort="30" lower="0" upper="1.309" velocity="1"/>
	</joint>
	<link name="finger_segment:11">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006181 -0.015507 -0.007416"/>
			<material name="finger_segment:11_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006181 -0.015507 -0.007416"/>
		</collision>
		<inertial>
			<mass value="0.000009"/>
			<inertia ixx="0" ixy="0" ixz="-0" iyy="0" iyz="0" izz="0"/>
			<origin rpy="0 0 0" xyz="0.005409 0.000556 -0.005751"/>
		</inertial>
	</link>
	<joint name="r_thumb_1st_x" type="revolute">
		<parent link="finger_segment:21"/>
		<child link="finger_segment:11"/>
		<origin rpy="-0.095246 0 0" xyz="-0.0005 0 -0.016"/>
		<axis xyz="-1 0 0"/>
		<limit effort="30" lower="0" upper="1.309" velocity="1"/>
	</joint>
	<link name="fingertip:11">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/fingertip_ipt_8417b8a9.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.005514 0.000374 -0.010171"/>
			<material name="fingertip:11_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/fingertip_ipt_8417b8a9.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.005514 0.000374 -0.010171"/>
		</collision>
		<inertial>
			<mass value="0.000008"/>
			<inertia ixx="0" ixy="-0" ixz="-0" iyy="0" iyz="-0" izz="0"/>
			<origin rpy="0 0 0" xyz="0.005013 0.000408 -0.010216"/>
		</inertial>
	</link>
	<joint name="r_thumb_2nd_x" type="revolute">
		<parent link="finger_segment:11"/>
		<child link="fingertip:11"/>
		<origin rpy="-0.109614 0 0" xyz="0 0 -0.016"/>
		<axis xyz="-1 0 0"/>
		<limit effort="30" lower="0" upper="1.309" velocity="1"/>
	</joint>
	<link name="finger_segment:23">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006681 -0.015507 -0.007416"/>
			<material name="finger_segment:23_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006681 -0.015507 -0.007416"/>
		</collision>
		<inertial>
			<mass value="0.000009"/>
			<inertia ixx="0" ixy="0" ixz="-0" iyy="0" iyz="0" izz="0"/>
			<origin rpy="0 0 0" xyz="0.004909 0.000556 -0.005751"/>
		</inertial>
	</link>
	<joint name="r_indexfinger_base_x" type="revolute">
		<parent link="right_palm:11"/>
		<child link="finger_segment:23"/>
		<origin rpy="-0.578103 0.062904 -3.141593" xyz="-0.001139 0.006226 -0.040003"/>
		<axis xyz="-0.998022 -0.00962907 -0.0621208"/>
		<limit effort="30" lower="0" upper="1.309" velocity="1"/>
	</joint>
	<link name="finger_segment:13">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006181 -0.015507 -0.007416"/>
			<material name="finger_segment:13_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/finger_segment_ipt_50d2a7f4.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.006181 -0.015507 -0.007416"/>
		</collision>
		<inertial>
			<mass value="0.000009"/>
			<inertia ixx="0" ixy="0" ixz="-0" iyy="0" iyz="0" izz="0"/>
			<origin rpy="0 0 0" xyz="0.005409 0.000556 -0.005751"/>
		</inertial>
	</link>
	<joint name="r_indexfinger_1st_x" type="revolute">
		<parent link="finger_segment:23"/>
		<child link="finger_segment:13"/>
		<origin rpy="-0.095246 0 0" xyz="-0.0005 0 -0.016"/>
		<axis xyz="-0.998022 -0.00962907 -0.0621208"/>
		<limit effort="30" lower="0" upper="1.309" velocity="1"/>
	</joint>
	<link name="fingertip:13">
		<visual>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/fingertip_ipt_8417b8a9.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.005514 0.000374 -0.010171"/>
			<material name="fingertip:13_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="package://sandbox//urdf/hand_test/fingertip_ipt_8417b8a9.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.005514 0.000374 -0.010171"/>
		</collision>
		<inertial>
			<mass value="0.000008"/>
			<inertia ixx="0" ixy="-0" ixz="-0" iyy="0" iyz="-0" izz="0"/>
			<origin rpy="0 0 0" xyz="0.005013 0.000408 -0.010216"/>
		</inertial>
	</link>
	<joint name="r_indexfinger_2nd_x" type="revolute">
		<parent link="finger_segment:13"/>
		<child link="fingertip:13"/>
		<origin rpy="-0.109614 0 0" xyz="0 0 -0.016"/>
		<axis xyz="-0.998022 -0.00962907 -0.0621208"/>
		<limit effort="30" lower="0" upper="1.309" velocity="1"/>
	</joint>
</robot>

